src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -475,6 +479,8 @@ void DxFontRenderData::InitializeFeatureMap()
 
 // Routine Description:
 // - Updates our internal map of font features with the given features
+// - NOTE TO CALLER: Make sure to call UpdateFont after calling this for the feature changes

```

Comment:
If that is required, why doesn't this function just call UpdateFont? Is it because it will be returning information the caller needs to pick up?

Reply:
Hopefully this slightly long-winded explanation makes sense:

Ideally, setting the font features and axes should be done via `UpdateFont`. Unfortunately, `UpdateFont` is shared with conhost so we don't want to put features/axes in there because conhost is never going to use it, so it'll be very wasteful considering conhost is the majority use case. 

So, to get around all that, setting the font axes and features are separate functions, but callers should call update font afterwards. `UpdateFont` is kinda expensive, so I'd prefer we don't call it in both `SetFeatures` and `SetAxes`.

Reply:
It does make sense. I was thinking that it could be something like where `UpdateFont` just had optional parameters at the end that defaulted to `nullptr` for all the renderers except this one which would unify it all up into a single call ensuring that the update happens along with setting the properties, eliminates the need to pick out the specific derived type for the DX renderer, and would mean no changes really for the GDI and other renderers as they would default those parameters out. But if that still sounds crazy, then we'll punt until @lhecker and @skyline75489 get their hands all over refactoring the interface like they are eager to do.

Reply:
Okay the latest commit does that, let me know if you prefer it and if not its an easy enough revert!